core/src/main/java/org/bitcoinj/core/RejectMessage.java

```diff
@@ -112,6 +112,18 @@ public RejectMessage(RejectCode code, @Nullable Sha256Hash rejectedMessageHash,
         this.rejectedMessageHash = rejectedMessageHash;
     }
 
+    @Override
+    public int messageSize() {
+        int rejectedMessageLength = rejectedMessage.getBytes(StandardCharsets.UTF_8).length;
+        int size = VarInt.sizeOf(rejectedMessageLength) + rejectedMessageLength;
+        size++; // code
+        int reasonLength = reason.getBytes(StandardCharsets.UTF_8).length;
+        size += VarInt.sizeOf(reasonLength) + reasonLength;
+        if ("block".equals(rejectedMessage) || "tx".equals(rejectedMessage))
+            size += Sha256Hash.LENGTH;
+        return size;
+    }
```

Comment:
This one is a little trickier, but still could be written in a more functional style.

Reply:
Except for the terniary this is done.